packages/multiselect/src/react/__specs__/index.spec.tsx

```diff
@@ -33,6 +45,97 @@ describe('MultiSelectField', () => {
       expect(results).toHaveNoViolations()
     })
   })
+
+  describe('Basic story', () => {
+    const { Basic } = stories
+
+    it('should focus input when label clicked', () => {
+      render(<Basic {...(Basic.args as any)} />)
+      const [label] = screen.getAllByLabelText('The label')
+      const input = screen.getByRole('textbox')
+
+      userEvent.click(label)
+
+      expect(input).toHaveFocus()
+    })
+
+    it('should open the menu when input receives focus', async () => {
+      render(<Basic {...(Basic.args as any)} />)
+      const input = screen.getByRole('textbox')
+
+      userEvent.click(input)
+
+      const menu = screen.getByRole('listbox')
+      expect(input).toHaveFocus()
+      expect(menu).toBeVisible()
+    })
+
+    it('should close the menu when input loses focus', async () => {
+      render(<Basic {...(Basic.args as any)} />)
+      const input = screen.getByRole('textbox')
+
+      userEvent.click(input)
+      const menu = screen.getByRole('listbox')
+
+      expect(input).toHaveFocus()
+      expect(menu).toBeVisible()
+
+      userEvent.tab()
+
+      expect(input).not.toHaveFocus()
+      expect(menu).not.toBeVisible()
+    })
+
+    it('should close the menu when esc pressed', async () => {
+      render(<Basic {...(Basic.args as any)} />)
+      const input = screen.getByRole('textbox')
+
+      userEvent.click(input)
+      const menu = screen.getByRole('listbox')
+
+      expect(input).toHaveFocus()
+      expect(menu).toBeVisible()
+
+      userEvent.type(input, '{esc}')
+
+      expect(input).toHaveFocus()
+      expect(menu).not.toBeVisible()
+    })
+
+    it('should open the menu when alt+up pressed', async () => {
```

Comment:
Would you please reference the WCAG guideline so I can see the guidance around binding "alt+up/down" for menus?

Reply:
downshift provides all the WCAG needed interactions but @sweeting and I didn't feel like it was complete so we've enhanced the interaction following this component library's docs: https://www.telerik.com/kendo-react-ui/components/dropdowns/multiselect/keyboard-navigation/

here is their a11y statement https://www.telerik.com/kendo-react-ui/components/dropdowns/multiselect/accessibility/

Reply:
Would you say that MultiSelect follows the combobox pattern?

If so, looking at the WCAG recs here:
https://www.w3.org/TR/wai-aria-practices-1.2/examples/combobox/combobox-autocomplete-list.html

Looks like the menu will pop open on regular down arrow if it's not open.  That seems both similar and simpler.

Would that feel complete?  cc @sweeting

Reply:
up and down arrow are already used when the menu is **open** - they move selection of the focused item.

`alt+up` is really the only new interaction which closes the menu when `up` isn't a viable option. looking at the description of this test i might have gotten up/down switched up

Reply:
here is the standard. they are optional events

https://www.w3.org/TR/wai-aria-practices/#keyboard-interaction

Reply:
here is the less readable reference https://www.w3.org/TR/wai-aria-practices/#textbox-keyboard-interaction

alt+up/down are option i guess 🤷

Reply:
I kinda feel this thing this is both a comboxbox and pills. 
So really two components composited together. 

https://www.lightningdesignsystem.com/components/pills/#Accessibility
https://www.lightningdesignsystem.com/components/combobox/#Keyboard-Interaction

The whole alt+up is not really something I'm familiar with. Did we check with Amber to see if that was a common interaction pattern. Also I kinda like some of the rules found here and applying them to the pills.

Reply:
yes it's very similar to the [`TagsInput`(pills) component](https://github.com/pluralsight/design-system/blob/master/packages/tagsinput/src/react/index.tsx#L71). the `TagsInput` component uses downshift [useMultipleSelect](https://www.downshift-js.com/use-multiple-selection) but does not employ a combobox so there is a large enough difference that i separated them initially. this is the [PR](https://github.com/pluralsight/design-system/pull/1597) that introduced the TagsInput and MultiSelect components

some references you can explore if you're interested.
- here is the [downshift guide](https://www.downshift-js.com/use-multiple-selection#usage-with-combobox) that provided the initial implementation
- [here](https://github.com/microsoft/sonder-ui/tree/master/src/components/multiselect) is one of the best implementations i had initially research with an accompanying [article testing various approaches](https://www.24a11y.com/2019/select-your-poison-part-2/)

As for the keybindings: [here](https://www.w3.org/TR/wai-aria-practices/#textbox-keyboard-interaction) are the combobox keyboard interaction docs and they explain the alt+up/down but they are marked as optional. I think running it by amber is a great idea - i'll do that now

Reply:
amber gives the 👍 . her response

> Yes! I use it ALL the time, even when I'm not testing for a11y!